modules/tinymce/src/core/main/ts/api/util/I18n.ts

```diff
@@ -19,6 +19,11 @@ export type Untranslated = Primitive | TokenisedString | RawString | null | unde
 
 export type TranslatedString = string;
 
+const isDuplicated = (items: string[], item: string) => {
+  let count = 0;
+  return Arr.exists(items, (x) => (x === item ? ++count : count) > 1);
+};
```

Comment:
This just feels slow. Here's a cool trick that [I found on google](https://www.jsowl.com/check-for-duplicate-strings-in-an-array-in-javascript/):

```suggestion
const isDuplicated = (items: string[], item: string) => items.indexOf(item) !== items.lastIndexOf(item);
```

Reply:
If I'm not wrong, my algorithm stops when it has encountered a duplicate when searching through the array sequentially, while your algorithm checks through the array from the front and back. So I feel the speed is roughly the same but depends on the order of the array (unless browsers have implemented some optimisations for `indexOf`).

I feel that in most cases the duplicate keys will be placed next to each other so searching through the array sequentially will yield a faster result (assuming the browser doesn't arbitrarily reorder the keys when calling `Object.keys`).

What do you reckon?

Reply:
What I'm trying to avoid is using a function to do `===` instead of leveraging the underlying `indexOf` implementation. It would take a gigantic array, with duplicates very close together, for your implementation to be faster. Unless you feel like switching to a `for` loop, those are well optimised :)

The other option (also from that link) is something like

```
const afterFound = items.indexOf(item) + 1;
return items.indexOf(item, afterFound) > afterFound;
```

Which I think will achieve what you're after, continue searching sequentially for duplicates.

Reply:
Yes, this one would be a more optimised version of my idea using the `indexOf` method, thanks for the suggestion.